server/src/test/java/org/apache/cassandra/sidecar/handlers/NativeUpdateHandlerTest.java

```diff
@@ -0,0 +1,153 @@
+    Vertx vertx;
+    Server server;
+    StorageOperations mockStorageOperations = mock(StorageOperations.class);
+
+    @BeforeEach
+    void before() throws InterruptedException
+    {
+        Injector injector;
+        Module testOverride = Modules.override(new TestModule()).with(new NativeUpdateHandlerTest.NativeUpdateHandlerTestModule());
+        injector = Guice.createInjector(Modules.override(SidecarModules.all()).with(testOverride));
+        vertx = injector.getInstance(Vertx.class);
+        server = injector.getInstance(Server.class);
+        VertxTestContext context = new VertxTestContext();
+        server.start().onSuccess(s -> context.completeNow()).onFailure(context::failNow);
+        context.awaitCompletion(5, TimeUnit.SECONDS);
+    }
+
+    @AfterEach
+    void after() throws InterruptedException
+    {
+        CountDownLatch closeLatch = new CountDownLatch(1);
+        server.close().onSuccess(res -> closeLatch.countDown());
+        if (closeLatch.await(60, TimeUnit.SECONDS)) LOGGER.info("Close event received before timeout.");
+        else LOGGER.error("Close event timed out.");
+    }
+
+    @Test
+    void testStartNative(VertxTestContext ctx)
+    {
+        WebClient client = WebClient.create(vertx);
+        String payload = "{\"state\":\"start\"}";
+        client.put(server.actualPort(), "127.0.0.1", "/api/v1/cassandra/native").expect(ResponsePredicate.SC_ACCEPTED).sendBuffer(io.vertx.core.buffer.Buffer.buffer(payload), ctx.succeeding(resp -> {
+            verify(mockStorageOperations, times(1)).startNativeTransport();
+            ctx.completeNow();
+        }));
+    }
+
+    @Test
+    void testStopNative(VertxTestContext ctx)
+    {
+        WebClient client = WebClient.create(vertx);
+        String payload = "{\"state\":\"stop\"}";
+        client.put(server.actualPort(), "127.0.0.1", "/api/v1/cassandra/native").expect(ResponsePredicate.SC_ACCEPTED).sendBuffer(io.vertx.core.buffer.Buffer.buffer(payload), ctx.succeeding(resp -> {
+            verify(mockStorageOperations, times(1)).stopNativeTransport();
+            ctx.completeNow();
+        }));
+    }
+
+    @Test
+    void testInvalidState(VertxTestContext ctx)
+    {
+        WebClient client = WebClient.create(vertx);
+        String payload = "{\"state\":\"foo\"}";
+        client.put(server.actualPort(), "127.0.0.1", "/api/v1/cassandra/native").expect(ResponsePredicate.SC_BAD_REQUEST).sendBuffer(io.vertx.core.buffer.Buffer.buffer(payload), ctx.succeeding(resp -> {
+            ctx.completeNow();
+        }));
+    }
+
+
+    /**
+     * Test guice module for Node Decommission handler tests
+     */
+    class NativeUpdateHandlerTestModule extends AbstractModule
+    {
+        @Provides
+        @Singleton
+        public InstancesMetadata instanceMetadata()
+        {
+            final int instanceId = 100;
+            final String host = "127.0.0.1";
+            final InstanceMetadata instanceMetadata = mock(InstanceMetadata.class);
```

Comment:
NIT, omit final modifier for local fields
```suggestion
            int instanceId = 100;
            String host = "127.0.0.1";
            InstanceMetadata instanceMetadata = mock(InstanceMetadata.class);
```